src/main/java/com/maddyhome/idea/vim/extension/replacewithregister/ReplaceWithRegister.kt

```diff
@@ -148,6 +149,22 @@ private fun doReplace(editor: Editor, context: DataContext, caret: ImmutableVimC
 
   var usedType = savedRegister.type
   var usedText = savedRegister.text
+
+  val selection = visualSelection.caretsAndSelections.values.first()
+  val startOffset: Int = selection.vimStart
+  val endOffset: Int = selection.vimEnd + 1
+
+  val isEmptySelection: Boolean = startOffset == endOffset
```

Comment:
Don’t you think this logic is a little strange?
Usually, the start offset is lower than the end offset. I'd expect that incrementing the end offset by one will make the difference between the start offset and the end offset even bigger, but here we expect them to become the same value.